lib/journaled/transaction_ext.rb

```diff
@@ -0,0 +1,82 @@
+require 'active_record/connection_adapters/abstract/transaction'
+
+module Journaled
+  module TransactionExt
+    def before_commit_records
+      @_journaled_committing_records = true
+      super
+    end
+
+    def _journaled_transaction_handler
+      if @_journaled_transaction_handler&.active?
+        @_journaled_transaction_handler
+      else
+        @_journaled_transaction_handler = TransactionHandler.new(txn: self).tap do |txn_handler|
+          txn_handler.joinable = !@_journaled_committing_records
```

Comment:
Defaulting to being non-joinable if we're already in a commit callback for the current transaction.